cli/fathom_web/commands/train.py

```diff
@@ -54,12 +54,34 @@ def learn(learning_rate, iterations, x, y, confidence_threshold, num_prunes, val
     if stopped_early:
         print(f'Stopping early at iteration {t}, just before validation error rose.')
 
+    print(f'Successfully trained model using threshold {confidence_threshold}.')
+    optimal_thresholds, max_accuracy = find_optimal_threshold(y, y_pred, num_prunes, threshold_incr=0.05)
+
+    print(f'Suggested threshold(s) are: {optimal_thresholds}.')
+    print(f'All listed thresholds have the same accuracy value of {max_accuracy}')
+
     # Horizontal axis is what confidence. Vertical is how many samples were that confidence.
     writer.add_histogram('confidence', confidences(model, x), t)
     writer.close()
     return model
 
 
+def find_optimal_threshold(y, y_pred, num_prunes, threshold_incr=0.05):
+    max_range = int(1.0 / threshold_incr) + 1
```

Comment:
It seems to me (and I could be wrong) that there's a better way to go at this than trying a set of static positions in [0, 1]. Accuracy changes only when we move the threshold to the other side of a datapoint. So all we really need to do is slide the accuracy past each of the datapoints in turn, measuring accuracy at each. And, for best expected results in an infinite universe of discourse, we'd put our final threshold halfway between the 2 datapoints it divides. This removes the need for a threshold and guarantees optimal accuracy, even if datapoints are closer together than `threshold_incr`. What do you think?

Reply:
For each datapoint?  You're not concerned with the performance?  How many datapoints are usually expected?

Reply:
(Replying from my phone didn't maintain the thread structure, so I'm moving this in-thread by hand.)

So far we’ve had thousands-ish, and maybe we can vectorize. I’d much rather have a better model than faster training. Though, if it takes seconds, we should just go with what you've got. I expect the exhaustive search will be O(n^2), given that we'd have to evaluate every datapoint for every threshold position, unless we get clever. I don't think we could straight binary-search, but there's probably some pruning to be had. Regardless, we shouldn't get fancy unless we need to. :-)